exporters/otlp/internal/otlpconfig/options.go

```diff
@@ -64,6 +65,9 @@ type (
 		Timeout     time.Duration
 		URLPath     string
 
+		// http configurations
+		HTTPTransport http.RoundTripper
```

Comment:
If we are going to have our own RoundTripper, then we should migrate the other options into it, so there is only one source of truth.

I can see `Insecure`, `TLSCfg`, `Compression`, and `Timeout` all being candidates.